db_test.go

```diff
@@ -900,6 +902,96 @@ func TestWithContextDoesNotMutate(t *testing.T) {
 	}
 }
 
+func TestWithDeadline(t *testing.T) {
+	db := makeTestDB(t, usersDDL)
+	db.DeadlineQueryRewriter = func(db *DB, query interface{}, millis int64) (interface{}, error) {
+		serializer, err := db.GetSerializer(query)
```

Comment:
I'm imagining that some might want to delve into the query AST, although perhaps it's better to just make it a private field for now and add setter functions as we identify different strategies for different database engines.